core/src/main/java/hudson/WebAppMain.java

```diff
@@ -222,8 +225,17 @@ public Locale get() {
                 public void run() {
                     boolean success = false;
                     try {
-                        Jenkins instance = new Hudson(_home, context);
-                        context.setAttribute(APP, instance);
+                        // need this for _home and other initialization
+                        Jenkins instance = new SetupWizard(_home, context);
+
+                        // Start immediately with the setup wizard for new installs
+                        if (InstallState.NEW.equals(InstallUtil.getInstallState())) {
+                            context.setAttribute(APP, instance);
+                        } else {
+                            instance.cleanUp(); // or we'll have "multiple Jenkins"
+                            instance = new Hudson(_home, context);
+                            context.setAttribute(APP, instance);
```

Comment:
This happens either way, so no need to have it inside either branch of the `if`, right?